hr_expense_from_purchase_request/models/hr_expense.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class HrExpenseSheet(models.Model):
+    _inherit = "hr.expense.sheet"
+
+    employee_user_id = fields.Many2one(
+        related="employee_id.user_id",
+        readonly=True,
+    )
+    purchase_request_id = fields.Many2one(
+        comodel_name="purchase.request",
+        string="Purchase Request",
+        ondelete="restrict",
+        copy=False,
+        index=True,
+        help="Select purchase request of this employee, to create expense lines",
+    )
+    pr_for = fields.Selection(
+        selection=[("expense", "Expense")],
+        string="Use PR for",
+        default="expense",
+        required=True,
+    )
+    pr_line_ids = fields.One2many(
+        comodel_name="hr.expense.sheet.prline",
+        inverse_name="sheet_id",
+        copy=False,
+    )
+
+    @api.onchange("employee_id")
+    def _onchange_purchase_request_employee(self):
+        self.purchase_request_id = False
+
+    @api.onchange("purchase_request_id")
+    def _onchange_purchase_request_id(self):
+        SheetPRLine = self.env["hr.expense.sheet.prline"]
+        self.pr_line_ids = False
+        for line in self.purchase_request_id.line_ids:
+            sheet_prline = self._prepare_sheet_prline(line)
+            self.pr_line_ids += SheetPRLine.new(sheet_prline)
+
+    def _prepare_sheet_prline(self, line):
+        """ Prepare data, to create hr.expense. All must be hr.expense's fields """
+        unit_amount = (
+            line.estimated_cost / line.product_qty if line.product_qty > 0 else 0
+        )
+        return {
+            "name": line.name,
+            "product_id": line.product_id.id,
+            "product_uom_id": line.product_uom_id.id,
+            "unit_amount": unit_amount,
+            "quantity": line.product_qty,
+            "total_amount": unit_amount * line.product_qty,
```

Comment:
`total_amount` should be computed field, for my extension module `unit_amount` no readonly field. If I changed this one `total_amount` is not updated.